File: src/main/scala/org/xbery/artbeams/common/access/domain/EntityAccessCount.scala
package org.xbery.artbeams.common.access.domain

/**
  * Aggregated count of user accesses to an entity.
  *
  * @author Radek Beran
  */
@SerialVersionUID(-1L)
case class EntityAccessCount(
  /* Unique key of an accessed entity. */
  entityKey: EntityKey,
  /* Count of accesses. */
  count: Long
) extends Serializable

object EntityAccessCount {
  final val CacheName = "entityAccessCounts"

  lazy val Empty = EntityAccessCount(
    EntityKey.Empty,
    0
  )
}

File: src/main/kotlin/org/xbery/artbeams/common/access/domain/EntityAccessCount.kt
package org.xbery.artbeams.common.access.domain

import java.io.Serializable

/**
 * Aggregated count of user accesses to an entity.
 *
 * @author Radek Beran
 */
data class EntityAccessCount(
    /* Unique key of an accessed entity. */
    val entityKey: EntityKey,
    /* Count of accesses. */
    val count: Long
) : Serializable {
    companion object {
        const val CacheName = "entityAccessCounts"
        val Empty = EntityAccessCount(
                EntityKey.Empty,
                0
        )
    }
}
